libvast/vast/chunk.hpp

```diff
@@ -78,11 +78,18 @@ class chunk final : public caf::ref_counted {
   /// accidental copies when calling this function.
   /// @returns A chunk pointer or `nullptr` on failure.
   template <class Buffer>
-    requires(!(std::is_lvalue_reference_v<
-                 Buffer> || std::is_trivially_move_assignable_v<Buffer>))
-  static auto make(Buffer&& buffer) -> decltype(as_bytes(buffer), chunk_ptr{}) {
-    const auto view = as_bytes(buffer);
-    return make(view, [buffer = std::exchange(buffer, {})]() noexcept {
+    requires(!std::is_lvalue_reference_v<Buffer> &&          //
+             !std::is_trivially_move_assignable_v<Buffer> && //
```

Comment:
I think we can drop the `!is_trivially_move_assignable` requirement now, right?

Reply:
Good catch. I dropped the requirement in a separate commit.